packages/iov-bns/src/bnsconnection.spec.ts

```diff
@@ -1342,8 +1340,8 @@ describe("BnsConnection", () => {
       recipient: rcptAddr,
       amount: [
         {
-          whole: 21,
-          fractional: 0,
+          quantity: "21000000000",
```

Comment:
you know what would make these tests easier to read... a little helper function.

```ts
amount: [
  {
    ...toQuantity("21.0", 9),
    tokenTicker: cash
  }
]

function toQuantity(decimal: string, fractionalDigits: number): {quantity: string, fractionalDigits: number} {
   ....
}
```

Reply:
And yes, this helper is just meant for tests, not for exporting to library clients, but it would make the tests easier to read and write

Reply:
Does the whole and the factional split matter for that test? This is 21000000000 CASH atomics. I don't like indirection in general as it intoduces more potental for bugs.

But what we could do is allow "_" in out integer strings just as in pure JavaScript. The API contract would allow them and any function dealing with `Amount` inputs must strip them off.

Reply:
This would change parsing to

```ts
const whole = amount.replace("_", "").quantity.slice(0, -amount.fractionalDigits);
const fractional = amount.replace("_", "").quantity.slice(-amount.fractionalDigits);
```

Reply:
Sure, allowing _ in strings seems like an easy win.

Reply:
A little bit if readability for a little bit of added complexity. Like that tradeof? https://github.com/iov-one/iov-core/pull/565/commits/2374e9b1df83dac1ddc86c0f5858c674a0bbb7da

Reply:
This makes checks like `expect(tx).toEqual(sendTx);` invalid because `quantity` has multiple string represenations now that are all equal amounts (https://github.com/iov-one/iov-core/pull/565/commits/9dcce03a160ba6e6086fc95ff6e696558ad99cca).

Reply:
facepalm.

yeah, this can be annoying. maybe revert my suggestion and stick with the original.
whatever you find best.

Reply:
Yeah, will revert. I am afraid this was not the last surprise here